### PR TITLE
feat: display item status

### PR DESCRIPTION
**Pull request by mimoham24**
- Case1: Item is not published & not in any request
    - Show `Draft`
- Case2: Item is not published & in some requests (waiting)
    - Show `Review` or `Review in progress`
- Case3: Item is published & not in any request
    - Show `Public`
- Case4: Item is published & in some requests (waiting)
    - Show `Public` and `In review`
- Case5: Item is published & in some request whose status is `closed`
    - Show `Public`
- Case6: request is closed & item not in any other request whose status (public or review)
    - show draft

### CODE REVIEW

File: server/internal/infrastructure/mongo/request.go [hunk 1]
```diff
@@ -65,6 +65,17 @@ func (r *Request) FindByIDs(ctx context.Context, ids id.RequestIDList) (request.
...
 }
 
+func (r *Request) FindByItems(ctx context.Context, list id.ItemIDList) (request.List, error) {
+
+	filter := bson.M{
+		"items.item": bson.M{
+			"$in": list.Strings(),
+		},
+	}
+
```
Comment (rot1024) [inline server/internal/infrastructure/mongo/request.go:75]:
```suggestion
	filter := bson.M{
		"items.item": bson.M{
			"$in": list.Strings(),
		},
	}
```

Also add an index for `items.item`

File: server/schemas/item.graphql [hunk 2]
```diff
@@ -10,6 +10,7 @@ type Item implements Node {
   user: User
   schema: Schema!
   model: Model!
+  status: [ItemStatus!]!
```
Comment (rot1024) [inline server/schemas/item.graphql:13]: I don't sure that why one item has multiple status

File: server/internal/usecase/interfaces/request.go [hunk 3]
```diff
@@ -42,6 +42,7 @@ type RequestFilter struct {
 type Request interface {
 	FindByID(context.Context, id.RequestID, *usecase.Operator) (*request.Request, error)
 	FindByIDs(context.Context, id.RequestIDList, *usecase.Operator) (request.List, error)
+	FindByItems(context.Context, id.ItemIDList, *usecase.Operator) (map[id.ItemID]request.List, error)
```
Comment (rot1024) [inline server/internal/usecase/interfaces/request.go:45]:
```suggestion
	ItemStatus(context.Context, id.ItemIDList, *usecase.Operator) (map[id.ItemID]item.Status, error)
```

The part that calculates the status from the request is business logic. It should be handled at the use case or domain layer.

**Review by rot1024 — CHANGES_REQUESTED** [review]

File: server/schemas/item.graphql (hunk 2 above)
Comment (mimoham24) [reply server/schemas/item.graphql:13]: as mentioned in the description item status might be `public` and `in review`

File: server/schemas/item.graphql [hunk 4]
```diff
@@ -29,6 +30,12 @@ type VersionedItem {
   value: Item!
 }
 
+enum ItemStatus {
+  DRAFT
+  PUBLIC
+  REVIEW
+}
```
Comment (rot1024) [inline server/schemas/item.graphql:37]:
```suggestion
enum ItemStatus {
  DRAFT
  DRAFT_REVIEW
  PUBLIC
  PUBLIC_REVIEW
}
```

File: server/schemas/item.graphql (hunk 2 above)
Comment (rot1024) [reply server/schemas/item.graphql:13]:
```suggestion
  status: ItemStatus!
```

**Review by rot1024 — CHANGES_REQUESTED** [review]

**Review by rot1024 — APPROVED** [review]

**Review by issmail-basel — APPROVED** [review]

**Pull request merged**